src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -141,6 +143,15 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
       payload.identifier = transformIdentifier(widgetProps, step, payload.identifier as string);
     }
 
+    // Widget rememberMe feature stores the entered identifier in a cookie, to pre-fill the form on subsequent visits to page
+    if (features?.rememberMe) {
+      if (payload.identifier) {
+        setUsernameCookie(payload.identifier as string);
+      }
+    } else {
+      removeUsernameCookie();
+    }
```

Comment:
Should this go inside the block above on L142 guarded by `if ('identifier' in payload)`?

Reply:
Ah yeah good catch